### DISCUSSION TITLE
Update resource after created ?

### DISCUSSION THREAD

**Discussion by neviaumi** [pulumi/pulumi#9303] [Pulumi in Practice] [answered]
Hi all , i currently usage pulumi to create my deployment setup

What i need to do is 

- Create S3 bucket
- Create CloudFront and point it to S3 bucket
- Update S3 bucket CORS to enable from client that hosted in S3

It easily to first 2 step ,

but how about step 3?

code i currently have
```js
const bucket = new aws.s3.Bucket(`${namePrefix}-bucket`, {
    website: {
      indexDocument: 'index.html',
    },
  });

  new aws.s3.BucketPolicy('bucketPolicy', {
    bucket: bucket.bucket,
    policy: bucket.bucket.apply(publicReadPolicyForBucket),
  });

  const cloudFrontDistribution = new aws.cloudfront.Distribution({....});

// What should i do here?
// bucket.update({cors: ....})
```

**Comment by Frassle** [comment, marked as answer]
We wrote a [blog](https://www.pulumi.com/blog/exploring-circular-dependencies/) about this problem in January. It describes some workarounds you can do for now, and also discuss some ideas for systems we might add to support this in a better way.
